mypy checks fail with click==8.1.0

There is a [typing issue](https://github.com/pallets/click/issues/2227) with `click==8.1.0` which is causing the `mypy` check to fail in the `main` branch and https://github.com/googleapis/gapic-generator-python/pull/1251.  See build log [here](https://github.com/googleapis/gapic-generator-python/runs/5753213728?check_suite_focus=true).

```
Run nox -s mypy
nox > Running session mypy
nox > Creating virtual environment (virtualenv) using python3.10 in .nox/mypy
nox > python -m pip install mypy 'types-protobuf<=3.19.[7](https://github.com/googleapis/gapic-generator-python/runs/5753213728?check_suite_focus=true#step:5:7)' types-PyYAML types-dataclasses
nox > python -m pip install .
nox > mypy gapic
gapic/cli/generate.py:71: error: <nothing> not callable
Found 1 error in 1 file (checked 36 source files)
```

--- a/setup.py
+++ b/setup.py
@@ -43,7 +43,9 @@ setup(
     platforms="Posix; MacOS X",
     include_package_data=True,
     install_requires=(
-        "click >= 6.7",
+        # There is a typing issue with click==8.1.0
+        # See https://github.com/pallets/click/issues/2227
+        "click >= 6.7,!=8.1.0",
         "google-api-core >= 2.3.2",
         "googleapis-common-protos >= 1.54.0",
         "grpcio >= 1.24.3",

